### ISSUE TITLE
Remove deprecated IReactorTime.cancelCallLater

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4076]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4076|
|Type|task|
|Created|2009-11-10 18:51:48Z|
|Branch|https://github.com/twisted/twisted/tree/remove-cancelcalllater-4076|

```IReactorTime.cancelCallLater``` in [twisted/internet/interfaces.py](twisted/internet/interfaces.py) is marked as deprecated so let's get rid of it.

Attachments:

* [4076.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e04/e04483fa29ce2a62129c0e1624ac1b82fdd899e1/207008fc92d3dadc508716792a8c80d57fbac2c1.diff) (1952 bytes) - added by cyli on 2010-06-06 06:57:11Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4076 4076
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_remove_cancelcalllater_4076 branches/remove-cancelcalllater-4076
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1257879108000000 1257879108000000
changetime__1277261481000000 1277261481000000
version__None None
owner__ 
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli removed owner|
|-|-|

Looks like this was deprecated in r20273 (in 2007).  Furthermore, t.i.base.ReactorBase.cancelCallLater has been deprecated since r2566 (2002).  Removed both (none of the tests in twisted.test.test_internet.py seem to test for it.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

t.i.base.ReactorBase.cancelCallLater has a comment saying do not delete.  If that shouldn't be deleted, then the deprecated IReactorTime.cancelCallLater should probably also not be removed (close this ticket as won't fix?).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

For ease of reference, the comment is

# DO NOT DELETE THIS - this is documented in Python in a Nutshell, so we we can't get rid of it for a long time.

Python in a Nutshell v1 was published in 2003, and v2 was published in 2006. I don't own the book, but I just asked someone who does to see if cancelCallLater was in v2 and she couldn't find it. If someone can confirm that it's not actually in v2 then maybe we can still get rid of it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

I think it's probably fine to get rid of this now (note that just because it's fine to get rid of something doesn't mean we _have_ to get rid of it, though).  As jesstess points out, it's been a "long time" by now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

And no, I didn't forget to remove the _review_ keyword.  I didn't review the patch, I'm just commenting on whether the ticket itself is valid.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

If you search through the Python in a Nutshell book on O'Reilly Safari Books Online (http://my.safaribooksonline.com/0596100469) for 'cancelCallLater', no results are returned.  If you search for 'twisted' (http://my.safaribooksonline.com/0596100469/540?bookview=search&query=twisted) you will get a bunch of results - look at page 540 (seems to be a rundown of reactor methods) and page 681 (seems to be the "C" part of the index).  cancelCallLater does not seem to be listed on either page.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess commented|
|-|-|

(In [29348]) Branching to 'remove-cancelcalllater-4076'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

(In [29349]) Apply patch 4076.diff by cyli.

refs [#4076](https://github.com/twisted/twisted/issues/4076)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_11"></a>|@jesstess set owner to @cyli|
|-|-|

Thanks for the patch, cyli! Looks good and ready to merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_12"></a>|@cyli set status to `closed`|
|-|-|

(In [29351]) Merge remove-cancelcalllater-4076 : remove t.i.base.ReactorBase.cancelCallLater

Author: cyli
Reviewer: jesstess
Fixes: [#4076](https://github.com/twisted/twisted/issues/4076)

Remove deprecated twisted.internet.base.ReactorBase.cancelCallLater.  It does not seem to be in Python in a Nutshell 2.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
